.github/workflows/main.yml

```diff
@@ -18,25 +18,62 @@ jobs:
         with:
           python-version: '3.10'
 
-      - name: Install dependencies
+      - name: Cache Poetry install
+        uses: actions/cache@v2
+        with:
+          path: ~/.local
+          key: poetry-1.5.1
+
+      - uses: snok/install-poetry@v1
+        with:
+          version: 1.5.1
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      - name: Cache deps
+        id: cache-deps
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: pydeps-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Cache Homebrew packages
+        uses: actions/cache@v2
+        with:
+          path: /usr/local/Homebrew
+          key: brew-packages-${{ runner.os }}-${{ hashFiles('Brewfile.lock.json') }}
+          restore-keys: |
+            brew-packages-${{ runner.os }}-
+            brew-packages-
+
+      # Install dependencies. `--no-root` means "install all dependencies but not the project
+      # itself", which is what you want to avoid caching _your_ code. The `if` statement
+      # ensures this only runs on a cache miss.
+      - run: poetry install --no-interaction --no-root
+        if: steps.cache-deps.outputs.cache-hit != 'true'
+
+      # Now install _your_ project. This isn't necessary for many types of projects -- particularly
+      # things like Django apps don't need this. But it's a good idea since it fully exercises the
+      # pyproject.toml and makes sure that if you add things like console-scripts at some point
+      # they'll be installed and working.
+      - run: poetry install --no-interaction
+
+      # Install Tesseract if not restored from cache
+      - name: Install Tesseract
         run: |
-          pip install wheel
-          pip install -r requirements.txt
-          pip install -e .
+          if [ ! -d "/usr/local/Cellar/tesseract" ]; then
+            brew install tesseract
+          fi
 
-      - name: Check formatting with Black
+      - name: Run tests
         run: |
-          black --check --exclude "(src|alembic)/" .
+          poetry run python -c "import nltk; nltk.download('punkt')"
```

Comment:
This line will be unnecessary once this PR is merged https://github.com/MLDSAI/OpenAdapt/pull/306

Reply:
Merged

Reply:
resolved in https://github.com/MLDSAI/OpenAdapt/pull/289/commits/02fcb0f9daf6ac1e81eb24165c8ed897b90af2f9